wire-ios-data-model/Source/ConversationSearch/SearchConversationsUseCase.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireUtilities
+
+/// This use case takes grouped conversations as input and returns only non-empty groups of conversations
+/// where the provided search text matches.
+///
+/// The search text is compared to the name of conversations as well as the names of the conversation's participants.
+public struct SearchConversationsUseCase<ConversationContainer>: SearchConversationsUseCaseProtocol
```

Comment:
The name `SearchConversationsUseCase` might lead some (including myself) to think it's about searching within conversations (i.e their content). Perhaps a less ambiguous name could be  `FilterConversationsUseCase`.

Reply:
Thank you, I've changed it.